cli/dcoscli/util.py

```diff
@@ -86,6 +86,46 @@ def confirm_text(prompt, confirmation_text):
     return False
 
 
+def choicelist(choices, descriptions=[], msg='Please choose: '):
+    """
+    Allow to pick an item from a choicelist. For example :
+
+        Please choose a login provider for your linked cluster:
+        1) dcos-uid-password
+        2) saml-sp-initiated
+        (1-2): [...]
+
+    :param choices: choices for the list
+    :type choices: list of str
+    :param descriptions: descriptions for the choices
+    :type descriptions: list of str
+    :returns: the chosen item
+    :rtype: str
+    """
+
+    if not descriptions:
+        descriptions = choices
+
+    if len(choices) != len(descriptions):
+        msg = 'Count mismatch between choices and descriptions.'
+        raise errors.DCOSException(msg)
+
+    emitter.publish(msg)
+    for i, desc in enumerate(descriptions):
+        emitter.publish('{}) {}'.format(i+1, desc))
+
+    # Write to stdout directly not to have a new line.
+    sys.stdout.write('({}-{}): '.format(1, len(choices)))
```

Comment:
What are the implications of this as opposed to using `emitter`?

Reply:
The problem with emitter is that it puts a new line at the end, which would make it look like this : 
```
Please choose a login provider for your linked cluster:
1) dcos-uid-password
2) saml-sp-initiated
(1-2):
[waiting for input here...]
```
And I wanted the cursor to be right new to `(1-2)`.

Reply:
Sure. I know that emitter prints the extra line. I'm more curious about what *else* emitter does (other than print) that we are missing out on by not using it here.

Reply:
The emitter just calls `print()`, that's how that function behaves, though I could add some arguments and rewrite these 2 lines as : 

    print('({}-{}): '.format(1, len(choices)), end='', flush=True)

Reply:
`end` defaulting to `\n`

Reply:
for reference https://stackoverflow.com/questions/493386/how-to-print-without-newline-or-space

Reply:
I'd rather extend the `emitter.publish()` command to take an argument to not print a newline somehow. I don't like having multiple ways of printing to the terminal.

Reply:
So do you suggest that I add an `end` and `flush` arguments to the emitter abstract class?

The flat emitter is just an implementation of it, and the print handler is just the default handler for that flat emitter. I'm not sure what was the intent for having the emitter abstraction in the first place, but here I feel like we might push some implementation-specific concepts into the abstraction.

Reply:
As I said, I just don't like having multiple ways of writing out to the console. At some point someone added the emitter class to abstract `print()` away, and it feels weird to sprinkle code around the code base that bypasses that. I'd rather extend the abstraction to do the right thing.